src/protocol/boolean/solved_bits.rs

```diff
@@ -97,13 +99,23 @@ impl SolvedBits {
             #[allow(clippy::cast_possible_truncation)]
             let c = c.narrow(&BitOpStep::Step(i as usize));
             async move {
+                #[allow(clippy::cast_possible_truncation)]
                 ConvertShares::new(xor_shares)
                     .execute_one_bit(c, record_id, i as u8)
                     .await
             }
         });
 
-        try_join_all(futures).await
+        // Pad 0's at the end to return `F::Integer::BITS` long bits
+        let mut b_b = try_join_all(futures).await?;
+        #[allow(clippy::cast_possible_truncation)]
+        b_b.append(
+            &mut repeat(Replicated::ZERO)
```

Comment:
These 0'a are additive shares of 0 in `F`, but not random. I think it's okay though because this protocol leaks `b < p` anyway.